File: CafeEase/CafeEase.Model/Requests/UserUpdateRequest.cs
namespace CafeEase.Model.Requests
{
    public class UserUpdateRequest
    {
        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string Email { get; set; }

        public string Username { get; set; }
        public int RoleId { get; set; }
    }
}

File: CafeEase/UI/cafeease_mobile/lib/models/user_update_request.dart
import 'package:json_annotation/json_annotation.dart';
part 'user_update_request.g.dart';

@JsonSerializable()
class UserUpdateRequest {
  String? firstName;
  String? lastName;
  String? email;
  String? username;
  int? roleId;
  int? cityId;

  UserUpdateRequest(
      {this.firstName, this.lastName, this.email, this.username, this.roleId,this.cityId});

  factory UserUpdateRequest.fromJson(Map<String, dynamic> json) =>
      _$UserUpdateRequestFromJson(json);

  Map<String, dynamic> toJson() => _$UserUpdateRequestToJson(this);
}
